integration/e2e/api_test.go

```diff
@@ -56,111 +57,230 @@ func TestSearchTagValuesV2(t *testing.T) {
+	require.NoError(t, tempo.WaitSumMetrics(e2e.Equals(1), "tempo_ingester_blocks_cleared_total"))
+
+	// Assert no more on the ingester
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			callSearchTagValuesV2AndAssert(t, tempo, tc.tagName, tc.query, searchTagValuesV2Response{}, 0, 0)
+		})
+	}
+
+	// Wait to blocklist_poll to be completed
+	require.NoError(t, tempo.WaitSumMetricsWithOptions(e2e.Equals(1), []string{"tempodb_blocklist_length"}, e2e.WaitMissingMetrics))
+
+	// Assert tags on storage backend
+	now := time.Now()
+	start := now.Add(-2 * time.Hour)
+	end := now.Add(2 * time.Hour)
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			callSearchTagValuesV2AndAssert(t, tempo, tc.tagName, tc.query, tc.expected, start.Unix(), end.Unix())
+		})
+	}
+}
+
+func TestSearchTags(t *testing.T) {
+	s, err := e2e.NewScenario("tempo_e2e_tags")
+	require.NoError(t, err)
+	defer s.Close()
+
+	require.NoError(t, util.CopyFileToSharedDir(s, configAllInOneLocal, "config.yaml"))
+	tempo := util.NewTempoAllInOne()
+	require.NoError(t, s.StartAndWaitReady(tempo))
+
+	jaegerClient, err := util.NewJaegerGRPCClient(tempo.Endpoint(14250))
+	require.NoError(t, err)
+	require.NotNil(t, jaegerClient)
+
+	batch := makeThriftBatch()
+	require.NoError(t, jaegerClient.EmitBatch(context.Background(), batch))
+
+	// Wait for the traces to be written to the WAL
+	time.Sleep(time.Second * 3)
+	callSearchTagAndAssert(t, tempo, searchTagsResponse{TagNames: []string{"service.name", "x", "xx"}}, 0, 0)
+
+	callFlush(t, tempo)
+	time.Sleep(time.Second * 30)
+	callFlush(t, tempo)
+
+	callSearchTagAndAssert(t, tempo, searchTagsResponse{}, 0, 0)
+	// Assert no more on the ingester
+
+	// Wait to blocklist_poll to be completed
+	time.Sleep(time.Second * 2)
+	// Assert tags on storage backend
+	now := time.Now()
+	start := now.Add(-2 * time.Hour)
+	end := now.Add(2 * time.Hour)
+	callSearchTagAndAssert(t, tempo, searchTagsResponse{TagNames: []string{"service.name", "x", "xx"}}, start.Unix(), end.Unix())
 }
 
-func callSearchTagValuesAndAssert(t *testing.T, svc *e2e.HTTPService, tagName, query string, expected searchTagValuesResponse) {
+func TestSearchTagValues(t *testing.T) {
+	s, err := e2e.NewScenario("tempo_e2e_tags")
+	require.NoError(t, err)
+	defer s.Close()
+
+	require.NoError(t, util.CopyFileToSharedDir(s, configAllInOneLocal, "config.yaml"))
+	tempo := util.NewTempoAllInOne()
+	require.NoError(t, s.StartAndWaitReady(tempo))
+
+	jaegerClient, err := util.NewJaegerGRPCClient(tempo.Endpoint(14250))
+	require.NoError(t, err)
+	require.NotNil(t, jaegerClient)
+
+	batch := makeThriftBatch()
+	require.NoError(t, jaegerClient.EmitBatch(context.Background(), batch))
+
+	// Wait for the traces to be written to the WAL
+	time.Sleep(time.Second * 3)
+	callSearchTagValuesAndAssert(t, tempo, "service.name", searchTagValuesResponse{TagValues: []string{"my-service"}}, 0, 0)
+
+	callFlush(t, tempo)
+	time.Sleep(time.Second * 22)
```

Comment:
same comment for sleep here? I 22 seconds is a lot of time and it adds up to e2e test runtime.